device_backends/src/TriggerDistributor.cc

```diff
@@ -126,32 +132,38 @@ namespace ChimeraTK {
 
   /********************************************************************************************************************/
 
-  template<typename UserType>
-  boost::shared_ptr<AsyncNDRegisterAccessor<UserType>> TriggerDistributor::SubscriptionImplementor<
-      UserType>::subscribeTo(TriggerDistributor& triggerDistributor, RegisterPath name, size_t numberOfWords,
-      size_t wordOffsetInRegister, AccessModeFlags flags) {
-    auto catalogue = triggerDistributor._backend->getRegisterCatalogue(); // need to store the clone you get
-    const auto& backendCatalogue = catalogue.getImpl();
-    // This code only works for backends which use the NumericAddressedRegisterCatalogue because we need the
-    // interrupt description which is specific for those backends and not in the general catalogue.
-    // If the cast fails, it will throw an exception.
-    auto removeMe = backendCatalogue.getRegister(name);
-    const auto& numericCatalogue = dynamic_cast<const NumericAddressedRegisterCatalogue&>(backendCatalogue);
-    auto registerInfo = numericCatalogue.getBackendRegister(name);
-
-    // Find the right place in the distribution tree to subscribe
-    boost::shared_ptr<AsyncAccessorManager> distributor;
-    if(registerInfo.getDataDescriptor().fundamentalType() == DataDescriptor::FundamentalType::nodata) {
-      distributor =
-          triggerDistributor.getDistributorRecursive<VariableDistributor<std::nullptr_t>>(registerInfo.interruptId);
-    }
-    else {
-      distributor = triggerDistributor.getDistributorRecursive<TriggeredPollDistributor>(registerInfo.interruptId);
+  namespace detail {
+
+    template<typename UserType>
+    boost::shared_ptr<AsyncNDRegisterAccessor<UserType>> AsyncDataAdapterSubscriptionImplementor<UserType,
+        std::nullptr_t>::subscribeTo(TriggerDistributor<std::nullptr_t>& triggerDistributor, RegisterPath name,
+        size_t numberOfWords, size_t wordOffsetInRegister, AccessModeFlags flags) {
+      auto catalogue = triggerDistributor._backend->getRegisterCatalogue(); // need to store the clone you get
+      const auto& backendCatalogue = catalogue.getImpl();
+      // This code only works for backends which use the NumericAddressedRegisterCatalogue because we need the
+      // interrupt description which is specific for those backends and not in the general catalogue.
+      // If the cast fails, it will throw an exception.
+      auto removeMe = backendCatalogue.getRegister(name);
```

Comment:
What does this line do? `removeMe` isn't referenced anywhere... If it's aiming for a side-effect, maybe add a comment and add `[[maybe_unused]]` to avoid a warning.

Reply:
removed